src/components/layout/vertical-nav.tsx

```diff
@@ -43,32 +43,41 @@ const NavLink: FC<
   );
 };
 
+const companies = [
+  {
+    name: "Work Go",
+    domain: "app.workgo.com",
+    logo: "/logo.png",
+  },
+  {
+    name: "Google",
+    domain: "google.com",
+    logo: "/google.png",
+  },
+  {
+    name: "Facebook",
+    domain: "facebook.com",
+    logo: "/facebook.png",
+  },
+  {
+    name: "Amazon",
+    domain: "amazon.com",
+    logo: "/amazon.png",
+  },
+];
```

Comment:
We should make this `as const` and infer the TS type from this automatically.

It would look something like `const companies = [ ... ] as const`

then we can simply do

`type Company = typeof companies[number]`

Instead of companies, let's rename them to workspaces (also the type as well)

Reply:
I defined the workspaces array in a new ts file and exported the array and the corresponding inferred types (for both WorkspacesType and WorkspaceType) since I could not export workspaces from a component file. Let me know if there is a better way to go about this.